src/wayland-eglsurface.c

```diff
@@ -2474,25 +2440,19 @@ EGLBoolean wlEglSurfaceRef(WlEglDisplay *display, WlEglSurface *surface)
 
 void wlEglSurfaceUnref(WlEglSurface *surface)
 {
-    uint32_t i;
-
     surface->refCount--;
     if (surface->refCount > 0) {
         return;
     }
 
     wlEglMutexDestroy(&surface->mutexLock);
+    wlEglMutexDestroy(&surface->ctx.streamImagesMutex);
 
     if (!surface->ctx.isOffscreen) {
         wlEglMutexDestroy(&surface->mutexFrameSync);
         pthread_cond_destroy(&surface->condFrameSync);
     }
 
```

Comment:
We no longer have to clean up the stream images array, but maybe it would be a good idea to assert that the `wl_list` is empty? By inspection it looks like we always clean up the surface context properly but an assert here could be helpful catching any future issues.

Reply:
An assert is already added to the line 2560.